archive/100-theorems-list/81_sum_of_prime_reciprocals_diverges.lean

```diff
@@ -0,0 +1,281 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import topology.instances.ennreal
+import algebra.squarefree
+
+/-!
+# Divergence of the Prime Reciprocal Series
+This file proves Theorem 81 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+The theorem states that the sum of the reciprocals of all prime numbers diverges.
+The formalization follows Erdős's proof by upper and lower estimates.
+
+## References
+https://en.wikipedia.org/wiki/Divergence_of_the_sum_of_the_reciprocals_of_the_primes
+-/
+
+open_locale big_operators
+open_locale classical
+open filter finset
+
+lemma card_le_div_nat {n p : ℕ} (hp : 0 < p) : card {e ∈ range n | p ∣ (e + 1)} ≤ n / p :=
+begin
+  let f : ℕ → ℕ := λ e, (e + 1) / p - 1,
+  let Np := {e ∈ range n | p ∣ (e + 1)},
+
+  have hf : ∀ a : ℕ, a ∈ Np → f a ∈ range (n / p),
+  { intros a ha,
+    simp only [Np, sep_def, mem_filter, mem_range] at ha,
+    obtain ⟨han, ⟨w, hw⟩⟩ := ha,
+    simp only [f, mem_range],
+    have hnp : n / p ≥ 1,
+    { rw ← nat.div_self hp,
+      exact nat.div_le_div_right (show n ≥ p, by nlinarith [(show w > 0, by nlinarith)]) },
```

Comment:
I don't like the doubly-nested `nlinarith`; may I suggest this instead:
```suggestion
    { have : 0 < w := by nlinarith,
      rw ← nat.div_self hp,
      exact nat.div_le_div_right (by nlinarith) },
```

Reply:
Also, if you notice, I switched out the instances of `>` and `≥` for `<` and `≤`. `>` and `≥` are not really used in mathlib.